### DISCUSSION TITLE
Feature: Option to pass custom data from custom handshake handler to ws session

### DISCUSSION THREAD

**Discussion by PlainBane** [PrivateRookie/ws-tool#21] [Ideas]
Hello,
It would be nice to be able to do something like this: 
```
`pub fn custom_handshake_handler(
    req: http::Request<()>,
) -> Result<(http::Request<()>, http::Response<String>,---> binary_custom_data), WsError> {
    // Custom handshake logic
    // Example: JWT parsing and validation
    let custom_data = "some_data"; // Example: JWT claims data

    // Option to pass custom_data to websocket session if handshake is successful
    // And then accessing custom_data value in the ws session
    let server = ServerBuilder::async_accept(
                stream,
                custom_handshake_handler,       
                AsyncWsBytesCodec::factory,
     ------->   binary_custom_data              // Serde serializable binary data or something portable
            )
}`
```

Something similar to how it's done in [https://github.com/uNetworking/uWebSockets.js/blob/master/examples/UpgradeAsync.js](https://github.com/uNetworking/uWebSockets.js/blob/master/examples/UpgradeAsync.js)
Line: 40, 56

**Comment by PrivateRookie** [comment]
If I understand the code in the link correctly, ws-tool already provides a similar function。For the client, you can add a custom handshake protocol through the `extension`, `hander` and other methods of the client builder, and then check the resp in the second param which is a closure of the async_connect/connect function [doc](https://docs.rs/ws-tool/latest/ws_tool/struct.ClientBuilder.html#method.async_connect)
For the server, the async_accept/accept method allows passing in two closures, which are used to check the handshake and construct the message codec [doc](https://docs.rs/ws-tool/latest/ws_tool/struct.ServerBuilder.html)
